unittests/linuxliketests.py

```diff
@@ -256,6 +256,36 @@ def test_pkgconfig_gen_deps(self):
             content = f.read()
             self.assertNotIn('-lstat2', content)
 
+    def test_pkgconfig_relocatable(self):
+        '''
+        Test that it generates relocatable pkgconfig when module
+        option pkgconfig.relocatable=true.
+        '''
+        testdir_rel = os.path.join(self.common_test_dir, '44 pkgconfig-gen')
+        self.init(testdir_rel, extra_args=['-Dpkgconfig.relocatable=true'])
+
+        def check_pcfile(name, *, relocatable, levels=2):
+            with open(os.path.join(self.privatedir, name), encoding='utf-8') as f:
+                pcfile = f.read()
+                (self.assertIn if relocatable else self.assertNotIn)(
+                    f'prefix={os.path.join("${pcfiledir}", *(".." for _ in range(levels)))}\n',
+                    pcfile)
+
+        check_pcfile('libvartest.pc', relocatable=True)
+        check_pcfile('libvartest2.pc', relocatable=True)
+
+        self.wipe()
+        self.init(testdir_rel, extra_args=['-Dpkgconfig.relocatable=false'])
+
+        check_pcfile('libvartest.pc', relocatable=False)
+        check_pcfile('libvartest2.pc', relocatable=False)
+
+        self.wipe()
+        testdir_abs = os.path.join(self.common_test_dir, '251 pkgconfig relocatable with absolute path')
```

Comment:
Is there any reason to test this as a project test *and* as a unittest? That will run it once and simply check that it successfully configures and builds, and run it again inside unittest in order to check the contents of files.

Reply:
Sorry, I was a bit confused with how testing is structured in meson. Realized now that I should move this to `test cases/unit`.

Reply:
This is now in `test cases/unit/105` and should only be run for the unittest `AllPlatformTests.test_pkgconfig_relocatable`.